RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.dev.cc

```diff
@@ -0,0 +1,1634 @@
+          if (once_per_block(acc)) {
+            // Fill PFCluster-level information
+            int rhIdx = pfClusteringVars[pfClusteringVars[topoId].topoSeedOffsets()]
+                            .topoSeedList();  // i is the seed rechit index
+            int seedIdx = pfClusteringVars[rhIdx].rhIdxToSeedIdx();
+            clusterView[seedIdx].energy() = pfRecHits[rhIdx].energy();
+            clusterView[seedIdx].x() = pfRecHits[rhIdx].x();
+            clusterView[seedIdx].y() = pfRecHits[rhIdx].y();
+            clusterView[seedIdx].z() = pfRecHits[rhIdx].z();
+          }
+        } else if (nSeeds <= 400 && nRHTopo - nSeeds <= 1500) {
+          hcalFastCluster_multiSeedIterative(
+              acc, pfClusParams, topoId, nSeeds, nRHTopo, pfRecHits, pfClusteringVars, clusterView, fracView);
+        } else {
+          if (once_per_block(acc))
+            printf("Topo cluster %d has %d seeds and %d rechits. Will be processed in next kernel.\n",
+                   topoId,
+                   nSeeds,
+                   nRHTopo);
+        }
+      } else
+        return;
+    }
+  };
+
+  PFClusterProducerKernel::PFClusterProducerKernel(Queue& queue, const reco::PFRecHitHostCollection& pfRecHits)
+      : nSeeds(cms::alpakatools::make_device_buffer<uint32_t>(queue)),
+        globalClusterPos(
+            cms::alpakatools::make_device_buffer<Position4[]>(queue, blocksForExoticClusters * maxTopoInput)),
+        globalPrevClusterPos(
+            cms::alpakatools::make_device_buffer<Position4[]>(queue, blocksForExoticClusters * maxTopoInput)),
+        globalClusterEnergy(
+            cms::alpakatools::make_device_buffer<float[]>(queue, blocksForExoticClusters * maxTopoInput)),
+        globalRhFracSum(cms::alpakatools::make_device_buffer<float[]>(queue, blocksForExoticClusters * maxTopoInput)),
+        globalSeeds(cms::alpakatools::make_device_buffer<int[]>(queue, blocksForExoticClusters * maxTopoInput)),
+        globalRechits(cms::alpakatools::make_device_buffer<int[]>(queue, blocksForExoticClusters * maxTopoInput)) {
+    alpaka::memset(queue, nSeeds, 0x00);  // Reset nSeeds
+  }
+
+  void PFClusterProducerKernel::execute(Queue& queue,
+                                        const reco::PFClusterParamsDeviceCollection& params,
+                                        reco::PFClusteringVarsDeviceCollection& pfClusteringVars,
+                                        reco::PFClusteringEdgeVarsDeviceCollection& pfClusteringEdgeVars,
+                                        const reco::PFRecHitHostCollection& pfRecHits,
+                                        reco::PFClusterDeviceCollection& pfClusters,
+                                        reco::PFRecHitFractionDeviceCollection& pfrhFractions) {
+    const int nRH = pfRecHits->size();
+    const int threadsPerBlock = std::is_same_v<Device, alpaka::DevCpu> ? 32 : 256;
+    const int blocks = std::is_same_v<Device, alpaka::DevCpu> ? nRH : divide_up_by(nRH, threadsPerBlock);
```

Comment:
Why not simply `divide_up_by(nRH, threadsPerBlock)` in all cases ?

(after fixing all kernel to use `elements_with_stride`)